pkg/reconciler/revision/resources/pa.go

```diff
@@ -50,10 +50,18 @@ func MakePA(rev *v1alpha1.Revision) *av1alpha1.PodAutoscaler {
 			},
 			ProtocolType: rev.GetProtocol(),
 			Reachability: func() av1alpha1.ReachabilityType {
+				// If the Revision has failed to become Ready, then mark the PodAutoscaler as unreachable.
+				if cond := rev.Status.GetCondition(v1alpha1.RevisionConditionReady); cond != nil && cond.Status == corev1.ConditionFalse {
+					// As a sanity check, also make sure that we don't do this when a
+					// newly failing revision is marked reachable by outside forces.
+					if !rev.IsReachable() {
```

Comment:
Minor: I don't think this condition is actually tested?

Reply:
Done